pkg/operator/workloadcontroller/workload_controller.go

```diff
@@ -140,10 +146,39 @@ func (c ServiceCatalogAPIServerOperator) sync() error {
 		}
 		return nil
 	case operatorsv1.Removed:
+		// delete the owner references from the service bindings
+		if err := c.deleteOwnerRefFromServiceBindings(); err != nil && !apierrors.IsNotFound(err) {
+			return err
+		}
+
 		// TODO probably need to watch until the NS is really gone
 		if err := c.kubeClient.CoreV1().Namespaces().Delete(operatorclient.TargetNamespaceName, nil); err != nil && !apierrors.IsNotFound(err) {
 			return err
 		}
+
+		err := wait.Poll(1*time.Second, 2*time.Minute, func() (stop bool, err error) {
```

Comment:
Should we `PollImmediate` here so we don't have to wait the first second if it's already gone?
```suggestion
		err := wait.PollImmediate(1*time.Second, 2*time.Minute, func() (stop bool, err error) {
```

Reply:
make sense